util/cpb/main.go

```diff
@@ -128,3 +161,107 @@ func getMetadata() (m *metadata, err error) {
 
 	return
 }
+
+// copyIndex searches for the index in an image and then copies it to dest
+// it will create intermediate directories for dest if they do not exist
+func copyIndex(dest string) (err error) {
+	path := getIndexPath()
+	if path == "" {
+		err = fmt.Errorf("could not find index data in image")
+		return
+	}
+
+	dir := filepath.Dir(dest)
+	if err = os.MkdirAll(dir, os.ModePerm); err != nil {
+		return
+	}
+
+	in, err := os.Open(path)
+	if err != nil {
+		return
+	}
+	defer in.Close()
+	out, err := os.Create(dest)
+	if err != nil {
+		return
+	}
+	defer func() {
+		cerr := out.Close()
+		if err == nil {
+			err = cerr
+		}
+	}()
+
+	hash := sha256.New()
+	hashAndCopy := io.MultiWriter(hash, out)
+
+	if _, err = io.Copy(hashAndCopy, in); err != nil {
+		return
+	}
+	if err = out.Sync(); err != nil {
+		return
+	}
+
+	status := reconciler.CopyIndexResult{Digest: fmt.Sprintf("%x", hash.Sum(nil))}
+	b, err := json.Marshal(status)
+	if err != nil {
+		return
+	}
+	return ioutil.WriteFile(reconciler.CopyIndexStatusLocation, b, os.ModePerm)
+}
+
+// getIndexPath searches standard paths for a db file and then falls back to walk the fs.
+// it is possible to get a false positive if there is more than one sqlite file in the image,
+// but in practice this is unlikely.
+func getIndexPath() (found string) {
+	standardPaths := []string{"/database/index.db", "/bundles.db"}
+	for _, p := range standardPaths{
+		_, err := os.Stat(p)
+		if os.IsNotExist(err) {
+			fmt.Println(p, "does not exist")
+			continue
+		}
+		f, err := os.Open(p)
+		if err != nil {
+			fmt.Println("err opening: ", err)
+			continue
+		}
+		if err := f.Close(); err != nil {
+			continue
+		}
+		if db, err := sqlittle.Open(p); err == nil {
+			fmt.Println("found", p)
+			db.Close()
+			found = p
+			return
+		}
+	}
+
+	_ = filepath.Walk("/", func(path string, info os.FileInfo, err error) error {
+		if info == nil {
+			return nil
+		}
+		if info.IsDir() {
+			return nil
+		}
+		if !info.Mode().IsRegular() {
+			return nil
+		}
+		f, err := os.Open(path)
+		if err != nil {
+			fmt.Println("err opening: ", err)
```

Comment:
nit: should this be `fmt.Printf` instead? In general these prints like they should be `Printf` but since the argument is a string it works as-is as well